ns1/resource_datafeed.go

```diff
@@ -35,24 +37,29 @@ func dataFeedResource() *schema.Resource {
 }
 
 func dataFeedToResourceData(d *schema.ResourceData, f *data.Feed) {
+	configAdapterOut(f)
 	d.SetId(f.ID)
 	d.Set("name", f.Name)
 	d.Set("config", f.Config)
 }
 
-func resourceDataToDataFeed(d *schema.ResourceData) *data.Feed {
+func resourceDataToDataFeed(d *schema.ResourceData) (feed *data.Feed, e error) {
+	err := configAdapterIn(d)
 	return &data.Feed{
 		Name:     d.Get("name").(string),
 		SourceID: d.Get("source_id").(string),
 		Config:   d.Get("config").(map[string]interface{}),
-	}
+	}, err
```

Comment:
Typically we'd return a `nil` `*data.Feed` if an err is returned, no?  If the caller of this func doesn't check err for whatever reason, they could incorrectly try to use the returned pointer.

Reply:
Additionally, we probably want to wrap the error with a little more context about what failed.  The default error message is not going to be very useful for terraform users if they provided an invalid value for `test_id` and are troubleshooting what failed.